tests/chainer_tests/datasets_tests/test_sub_dataset.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+
+from chainer import datasets
+from chainer import testing
+
+
+class TestSubDataset(unittest.TestCase):
```

Comment:
I think it is better to split this class into three classes, namely, for SubDataset, for split_dataset, and for `get_cross_validation_dataset`.